--- app/frontend/genproto/farm_ng_proto/tractor/v1/resource.ts	2020-09-24 10:33:12.000000000 -0700
+++ protos/resource.patch.ts	2020-09-24 10:33:03.000000000 -0700
@@ -1,3 +1,4 @@
+import * as Long from 'long';
 /* eslint-disable */
 import { Writer, Reader } from 'protobufjs/minimal';
 import { Int64Value } from '../../../google/protobuf/wrappers';
@@ -25,7 +26,7 @@
   /**
    *  (Optional) The uncompressed size of the resource in bytes.
    */
-  length: number | undefined;
+  length: Long | undefined;
 }

 const baseResource: object = {
@@ -88,7 +89,7 @@
       message.contentType = "";
     }
     if (object.length !== undefined && object.length !== null) {
-      message.length = Number(object.length);
+      message.length = Long.fromNumber(object.length);
     } else {
       message.length = undefined;
     }
@@ -112,7 +113,7 @@
       message.contentType = "";
     }
     if (object.length !== undefined && object.length !== null) {
-      message.length = object.length;
+      message.length = object.length as Long;
     } else {
       message.length = undefined;
     }
